file.go

```diff
@@ -615,7 +616,7 @@ func (r *filePages) SeekToRow(rowIndex int64) (err error) {
 		if index < 0 {
 			return ErrSeekOutOfRange
 		}
-		_, err = r.section.Seek(pages[index].Offset-r.dataOffset, io.SeekStart)
+		_, err = r.section.Seek(pages[index].Offset-r.baseOffset, io.SeekStart)
```

Comment:
The offset was miscalculated when a dictionary page was present since the base isn't the data page offset but the dictionary page offset in that case.